### DISCUSSION TITLE
Investigating Frequency response and num time steps in Sionna RTX

### DISCUSSION THREAD

**Discussion by manustearling** [NVlabs/sionna#175] [General]
Hello,

I have two questions related to Sionna RTX.

**First question:**

I would like to know if we calculate the frequency response several times with the same users positions Tx/Rx, will we obtain the same result? Why, in this [code](https://colab.research.google.com/drive/1G9k1wQPbCiz1wZxpq3wZwmrOyghcFNeA?usp=sharing), do we in most cases obtain the same frequency response value between the first and second iteration (h_PP[0,0,:] == h_PP[1,0,:]) ==> True? Despite the fact that the seed is not fixed in the compute_paths function and the fact that the signals are not in line of sight, so there are diffraction phenomena,...

**Second question:**

When calculating Paths2CIR, is setting num_time_step = 1 and putting it in a for loop 10 times equivalent to setting num_time_step = 10 (as in this [code](https://colab.research.google.com/drive/1G9k1wQPbCiz1wZxpq3wZwmrOyghcFNeA?usp=sharing))? And will this also be less expensive in terms of calculation time? My aim is to create a large dataset, and it seems that putting a high num_time_step value it costs less computing time than in other cases...

Thank you so much for your help

**Comment by jhoydis** [comment]
Hi,

First of all, you might want to upgrade your code examples to Sionna v0.15 which had some breaking changes, e.g., the function Paths2CIR does not exist anymore. 

**Answer to first question:**

As long as you restrict your simulations to LOS, reflections, and diffraction, the outputs of multiple runs of the ``compute_paths`` method are identical as there is simply nothing random happening. In v0.14, it was possible to get slightly different results with the "stochastic" method as not all of the possible propagation paths were found. In v0.15, this is not the case anymore as ray directions are deterministic. Once you add scattering, however, you will get different outputs if you do not set  a random seed using``tf.random.set_seed()`` just before calling ``compute_paths``. The reason for this is that the scattered field is obtained through Monte Carlo integration, i.e., the directions of scattered rays are random.

**Answer to second question:**

In Sionna v0.15 you will need to call ``paths.apply_doppler`` prior to ``paths.cir`` to obtain time-evolution of the CIR due to movement of the transmitter and/or receiver.  There is no point in running ``paths.apply_doppler`` multiple times as the output is the same. It is important to understand that time-evolution is applied artificially to the computed paths. This is different from running ``compute_paths`` multiple times with different radio device positions. 

Hope that clarifies your questions.
